src/services/identity/NotificationsService.ts

```diff
@@ -0,0 +1,258 @@
+    })
+
+    if (newNotifications.length > 0)
+      return this.formatNotifications(newNotifications)
+
+    const mostRecentNotifications = await this.repository.findAll({
+      where: {
+        user_id: userId,
+      },
+      order: [
+        ["first_read_time", "DESC NULLS FIRST"],
+        ["priority", "ASC"],
+      ],
+      limit: NUM_RECENT_NOTIFICATIONS,
+      include: [
+        {
+          model: Site,
+          as: "site",
+          required: true,
+          include: [
+            {
+              model: Repo,
+              required: true,
+              where: {
+                name: siteName,
+              },
+            },
+          ],
+        },
+      ],
+    })
+
+    return this.formatNotifications(mostRecentNotifications)
+  }
+
+  async listAll({ siteName, userId }: { siteName: string; userId: string }) {
+    const notifications = await this.repository.findAll({
+      where: {
+        user_id: userId,
+      },
+      order: [
+        ["first_read_time", "DESC NULLS FIRST"],
+        ["priority", "ASC"],
+      ],
+      include: [
+        {
+          model: Site,
+          as: "site",
+          required: true,
+          include: [
+            {
+              model: Repo,
+              required: true,
+              where: {
+                name: siteName,
+              },
+            },
+          ],
+        },
+      ],
+    })
+
+    return this.formatNotifications(notifications)
+  }
+
+  async markNotificationsAsRead({
+    siteName,
+    userId,
+  }: {
+    siteName: string
+    userId: string
+  }) {
+    const siteMember = await this.siteMember.findOne({
+      where: { user_id: userId },
+      include: [
+        {
+          model: Site,
+          required: true,
+          include: [
+            {
+              model: Repo,
+              required: true,
+              where: {
+                name: siteName,
+              },
+            },
+          ],
+        },
+      ],
+    })
+    const readAtDate = new Date()
+    await this.repository.update(
+      {
+        firstReadTime: readAtDate,
+      },
+      {
+        where: {
+          site_member_id: siteMember?.id,
+          first_read_time: null,
+        },
+      }
+    )
+  }
+
+  async create({
+    siteName,
+    userId,
+    link,
+    notificationType,
+    notificationSourceUsername,
+  }: {
+    siteName: string
+    userId: string
+    link: string
+    notificationType: NotificationType
+    notificationSourceUsername: string
+  }) {
+    const siteMember = await this.siteMember.findOne({
+      where: { user_id: userId },
+      include: [
+        {
+          model: Site,
+          required: true,
+          include: [
+            {
+              model: Repo,
+              required: true,
+              where: {
+                name: siteName,
+              },
+            },
+          ],
+        },
+      ],
+    })
+    const recentTargetNotification = await this.repository.findOne({
```

Comment:
We reusing it because we don't want to create new notifications if the previous one is too recent, i've elaborated more on this in https://github.com/isomerpages/isomercms-backend/pull/508/commits/9c4f6233a930ce03a5f2c668deeef114737e81a5